scvi/model/base/_pyromixin.py

```diff
@@ -18,59 +17,17 @@
 logger = logging.getLogger(__name__)
 
 
-class PyroJitGuideWarmup(Callback):
-    """A callback to warmup a Pyro guide.
+def setup_pyro_model(dataloader, pl_module):
+    """Way to warmup Pyro Model and Guide in an automated way.
 
-    This helps initialize all the relevant parameters by running
-    one minibatch through the Pyro model.
+    Setup occurs before any device movement, so params are iniitalized on CPU.
     """
-
-    def __init__(self, dataloader: AnnDataLoader = None) -> None:
-        super().__init__()
-        self.dataloader = dataloader
-
-    def on_train_start(self, trainer, pl_module):
-        """Way to warmup Pyro Guide in an automated way.
-
-        Also device agnostic.
-        """
-        # warmup guide for JIT
-        pyro_guide = pl_module.module.guide
-        if self.dataloader is None:
-            dl = trainer.datamodule.train_dataloader()
-        else:
-            dl = self.dataloader
-        for tensors in dl:
-            tens = {k: t.to(pl_module.device) for k, t in tensors.items()}
-            args, kwargs = pl_module.module._get_fn_args_from_batch(tens)
-            pyro_guide(*args, **kwargs)
-            break
-
-
-class PyroModelGuideWarmup(Callback):
-    """A callback to warmup a Pyro guide and model.
-
-    This helps initialize all the relevant parameters by running
-    one minibatch through the Pyro model. This warmup occurs on the CPU.
-    """
-
-    def __init__(self, dataloader: AnnDataLoader) -> None:
-        super().__init__()
-        self.dataloader = dataloader
-
-    def setup(self, trainer, pl_module, stage=None):
-        """Way to warmup Pyro Model and Guide in an automated way.
-
-        Setup occurs before any device movement, so params are iniitalized on CPU.
-        """
-        if stage == "fit":
-            pyro_guide = pl_module.module.guide
-            dl = self.dataloader
-            for tensors in dl:
-                tens = {k: t.to(pl_module.device) for k, t in tensors.items()}
-                args, kwargs = pl_module.module._get_fn_args_from_batch(tens)
-                pyro_guide(*args, **kwargs)
-                break
+    for tensors in dataloader:
```

Comment:
Better to do next(iter(dataloader)) to get a single batch. I think still having the class makes sense. Within this class, there can be a manual_start function.